ui/src/app/features/drag-n-drop/drag-n-drop.component.ts

```diff
@@ -9,27 +27,79 @@ export class DragNDropComponent {
 
   @ViewChild('fileDropRef', { static: false }) fileDropEl: ElementRef;
   file: any;
+  data: object;
+  loading = true;
+  @Input() model: Model;
+
+  constructor(private dragService: DragNDropService) {}
+
+  /**
+   * on file drop handler
+   */
+  onFileDroppedAdd($event) {
+    this.processFileAddToModel($event);
+  }
 
   /**
    * on file drop handler
    */
-  onFileDropped($event) {
-    this.processFile($event);
+  onFileDroppedReco($event) {
+    this.processFileRecoFace($event);
+  }
+
+  /**
+   * handle file from browsing
+   */
+  fileBrowseHandlerAdd(files) {
+    this.processFileAddToModel(files);
   }
 
   /**
    * handle file from browsing
    */
-  fileBrowseHandler(files) {
-    this.processFile(files);
+  fileBrowseHandlerReco(files) {
+    this.processFileRecoFace(files);
   }
 
   /**
-   * Process file
+   * Add faces to model
    * @param files (Files List)
    * TODO Send file to api
    */
-  processFile(files: Array<any>) {
-    this.file = files[0];
+  processFileAddToModel(files: Array<any>) {
+    for (const item of files) {
+      this.file = item;
+    }
+    this.dragService.addFace(this.file, this.model).subscribe((res) => {
+      this.data = res;
+      this.loading = false;
+    });
   }
+
+  /**
+   * Recognize face
+   * @param files (Files List)
+   * TODO Send file to api
+   */
+  processFileRecoFace(files: Array<any>) {
+    for (const item of files) {
+      this.file = item;
+    }
+    console.log(files.length);
+    this.dragService.recognize(this.file, this.model).subscribe((res) => {
+      this.data = res;
+      this.loading = false;
+    });
+  }
+
+  /**
+   * Train model
+   */
+  trainModel() {
+    this.dragService.train(this.model).subscribe((res) => {
```

Comment:
same as above. The dump component should only emit value through Output. The container component should dispatch the action. API calls should be done in effects